src/shell/utils.js

```diff
@@ -163,6 +163,7 @@ function installService() {
         [`${confDir}/google-chrome-beta/NativeMessagingHosts/`, google],
         [`${confDir}/google-chrome-unstable/NativeMessagingHosts/`, google],
         [`${confDir}/BraveSoftware/Brave-Browser/NativeMessagingHosts/`, google],
+        [`${confDir}/BraveSoftware/Brave-Browser-Beta/NativeMessagingHosts/`, google],
         [`${confDir}/BraveSoftware/Brave-Browser-Nightly/NativeMessagingHosts/`, google],
         [`${homeDir}/.mozilla/native-messaging-hosts/`, mozilla],
         [`${homeDir}/.config/microsoft-edge-dev/NativeMessagingHosts`, google],
```

Comment:
Sounds like a better option to me. From the looks of the code, we can modify `util.js` to walk the `.config` directory and, as you mentioned, check if the directories exist. 

Also, this will eliminate the need for adding an entry for every browser in existence (the number is going up day by day, mostly chromium based)

Reply:
Something like this maybe, since Chromium browsers are really the problem case?

```js
const chromiumBrowsers = [
    `${confDir}/chromium`,
    `${confDir}/google-chrome`,
    `${confDir}/google-chrome-beta`,
    `${confDir}/google-chrome-unstable`,
    `${confDir}/BraveSoftware/Brave-Browser`,
    `${confDir}/BraveSoftware/Brave-Browser-Nightly`,
    `${confDir}/microsoft-edge-dev`,
    `${confDir}/microsoft-edge-beta`,
];

// Later...
for (const browserDir of chromiumBrowsers) {
    if (!GLib.file_test(browserDir, GLib.FileTest.IS_DIR))
        continue;

    _installFile(`{$browserDir}/NativeMessagingHosts`, manifestFile, google);
}

if (GLib.file_test(`${homeDir}/.mozilla`, GLib.FileTest.IS_DIR))
    _installFile(`${homeDir}/.mozilla/native-messaging-hosts`, manifestFile, mozilla);
```

I have a feeling a more complex approach might be a bit fragile, but that's just my two cents :) It's also worth considering this is performed every time `init()` is run, so the fewer I/O calls the better unless someone wants to create an asynchronous version of this.

Reply:
@andyholmes Something very much like that, yeah. It's at least a start, and a pretty good one.

It would at least mean I don't have two (soon to be three) totally superfluous Brave config dirs, and two equally useless Edge config dirs, in my `~/.config/`... when neither browser has ever been installed on my system in any capacity. (As well as Chromium and one of the three Chrome channels, though it's possible I already had at least vestigial configs for either or both of those.)

I'd still love to reach a point where this could somehow adapt to previously-unknown browser iterations without someone having to keep firing off PRs to manually insert _yet another_ path, every time the WebKit virus undergoes its weekly mutation and spawns a new strain. Maybe we won't, but I can dream.

And hey, it's at least a step _closer_ to that point, and progress is progress.